test/Pester/EditorServices.Integration.Tests.ps1

```diff
@@ -0,0 +1,78 @@
+
+$script:ExceptionRegex = [regex]::new('\s*Exception: (.*)$', 'Compiled,Multiline,IgnoreCase')
+function ReportLogErrors
+{
+    param(
+        [Parameter()][string]$LogPath,
+
+        [Parameter()][ref]<#[int]#>$FromIndex = 0,
+
+        [Parameter()][string[]]$IgnoreException = @()
+    )
+
+    $logEntries = Parse-PsesLog $LogPath |
+        Where-Object Index -ge $FromIndex.Value
+
+    # Update the index to the latest in the log
+    $FromIndex.Value = ($FromIndex.Value,$errorLogs.Index | Measure-Object -Maximum).Maximum
+
+    $errorLogs = $logEntries |
+        Where-Object LogLevel -eq Error |
+        Where-Object {
+            $match = $script:ExceptionRegex.Match($_.Message.Data)
+
+            (-not $match) -or ($match.Groups[1].Value.Trim() -notin $IgnoreException)
+        }
+
+    if ($errorLogs)
+    {
+        $errorLogs | ForEach-Object { Write-Error "ERROR from PSES log: $($_.Message.Data)" }
+    }
+}
+
+Describe "Loading and running PowerShellEditorServices" {
+    BeforeAll {
+        Import-Module -Force "$PSScriptRoot/../../module/PowerShellEditorServices"
+        Import-Module -Force "$PSScriptRoot/../../tools/PsesPsClient/out/PsesPsClient"
+        Import-Module -Force "$PSScriptRoot/../../tools/PsesLogAnalyzer"
+
+        $logIdx = 0
+        $psesServer = Start-PsesServer
+        $client = Connect-PsesServer -PipeName $psesServer.SessionDetails.languageServicePipeName
+    }
+
+    AfterAll {
+        try
+        {
+            $client.Dispose()
+            $psesServer.PsesProcess.Kill()
+            $psesServer.PsesProcess.Dispose()
+        }
+        catch
+        {
+            # Do nothing
+        }
+
+        # TODO: We shouldn't need to skip this error
+        ReportLogErrors -LogPath $psesServer.LogPath -FromIndex ([ref]$logIdx) -IgnoreException 'EndOfStreamException'
+    }
+
+    It "Starts and responds to an initialization request" {
+        $request = Send-LspInitializeRequest -Client $client
+        $response = Get-LspResponse -Client $client -Id $request.Id
+        $response.Id | Should -BeExactly $request.Id
+
+        ReportLogErrors -LogPath $psesServer.LogPath -FromIndex ([ref]$logIdx)
```

Comment:
Do you need to do this here and within the AfterAll?

Reply:
If so... maybe put it in an AfterEach instead?

Reply:
In an `AfterEach` it will crash all the remaining tests. Like this it will fail the test but keep going. I would have preferred an `AfterEach` but it just doesn't have the semantics we want.

Then the problem is the error we were seeing wasn't present until the `AfterAll`, which is why I had to stick it in there. Without that, no error on connection failure it seems

Reply:
I've actually moved the disposal logic to the last test now, since things make the *most* sense there (still a bit yucky, but it's better)